pkg/cdm/bootstrap.go

```diff
@@ -194,14 +194,28 @@ func (c *BootstrapClient) BootstrapCluster(ctx context.Context, nodeIP string, c
 	if err != nil {
 		return 0, fmt.Errorf("failed POST request %q: %s", endpoint, err)
 	}
-	if code != 202 {
-		return 0, fmt.Errorf("failed POST request %q: %s", endpoint, http.StatusText(code))
-	}
 
+	// We unmarshal the response from the bootstrap request early to capture
+	// potential error messages. Unmarshal errors are ignored at this time as
+	// certain responses could contain malformed JSON object.
 	var bootstrap struct {
-		ID int `json:"id"`
+		ID     int    `json:"id"`
+		Status string `json:"status"`
+	}
+	jsonErr := json.Unmarshal(buf, &bootstrap)
```

Comment:
Ought to be `err` right? (as that is what is being returned at line 219)

Reply:
Oh, disregard that.
But we should be returning `jsonErr` at line 219?

Reply:
I've updated the code accordingly.